### PR TITLE
Adding unit tests for ProductServiceImpl

### PR DESCRIPTION
**Pull request by Dauren-Delmukhambetov**
## Description

This pull request embraces changes on adding unit tests for `ProductServiceImpl` during the pair-programming session

### CODE REVIEW

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        when(mapper.toDto(new ProductEntity()))
+                .thenReturn(new Product());
+
+        final var product = productService.findById(123L);
+
+        assertThat(product).isNotNull();
+    }
+
+    @Test
+    void shouldThrowException_WhenUserNotFound() {
```
Comment (tdoff) [inline src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java:55]: It should be Product not User in test method's name

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        assertThrows(EntityNotFoundException.class, () -> productService.findById(99L));
+    }
+
+    @Test
+    void shouldThrowException_WhenUserIsDeleted() {
```
Comment (tdoff) [inline src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java:60]: the same change required

**Review by tdoff — CHANGES_REQUESTED** [review]
During pair programming I made mistake with method names ))

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java (hunk 1 above)
Comment (Dauren-Delmukhambetov) [inline src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java:55]:
```suggestion
    void shouldThrowException_WhenProductNotFound() {
```

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        final var entity = new ProductEntity();
+        entity.setDeletedAt(LocalDateTime.now());
+        when(repository.findById(99L))
+                .thenReturn(Optional.of(entity));
+        assertThrows(EntityDeletedException.class, () -> productService.findById(99L));
+    }
+}
```
Comment (Dauren-Delmukhambetov) [inline src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java:67]:
```suggestion
}

```

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java (hunk 2 above)
Comment (Dauren-Delmukhambetov) [inline src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java:60]:
```suggestion
    void shouldThrowException_WhenProductIsDeleted() {
```

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java (hunk 1 above)
Comment (Dauren-Delmukhambetov) [reply src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java]: Fixed

File: src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java (hunk 2 above)
Comment (Dauren-Delmukhambetov) [reply src/test/java/kz/toko/app/service/impl/ProductServiceImplTest.java]: Fixed

**Review by tdoff — APPROVED** [review]
Reviewed changes

**Pull request merged**